OsmAnd/src/net/osmand/plus/plugins/monitoring/VehicleMetricsRecordingFragment.java

```diff
@@ -0,0 +1,344 @@
+package net.osmand.plus.plugins.monitoring;
+
+import android.annotation.SuppressLint;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.DimenRes;
+import androidx.annotation.DrawableRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.StringRes;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.view.ViewCompat;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.fragment.app.FragmentManager;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.android.material.appbar.AppBarLayout;
+
+import net.osmand.plus.R;
+import net.osmand.plus.activities.MapActivity;
+import net.osmand.plus.base.BaseOsmAndFragment;
+import net.osmand.plus.helpers.AndroidUiHelper;
+import net.osmand.plus.settings.backend.preferences.ListStringPreference;
+import net.osmand.plus.utils.AndroidUtils;
+import net.osmand.plus.utils.ColorUtilities;
+import net.osmand.plus.views.mapwidgets.configure.dialogs.DistanceByTapFragment;
+import net.osmand.plus.widgets.dialogbutton.DialogButton;
+import net.osmand.shared.obd.OBDCommand;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class VehicleMetricsRecordingFragment extends BaseOsmAndFragment {
+
+	public static final String TAG = DistanceByTapFragment.class.getSimpleName();
+
+	private Toolbar toolbar;
+	private ImageView navigationIcon;
+
+	private RecyclerView recyclerView;
+	private VehicleMetricsRecordingAdapter adapter;
+
+	private final Set<String> selectedCommands = new HashSet<>();
+	private final List<Object> items = new ArrayList<>();
+	private DialogButton selectAllButton;
+	private boolean allCommandsSelected = false;
```

Comment:
It could be removed, just use selectedCommands.containsAll(getFlatListAvailableCommands())